### PR TITLE
Update ci.yaml to use new v4 Actions checkout & setup-java

### PR DESCRIPTION
**Pull request by thadguidry**
- this should eliminate the Node 16 -> 20 warnings

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "java.compile.nullAnalysis.mode": "automatic"
```
Comment (wetneb) [inline .vscode/settings.json:2]: Did you intend to commit this file? It looks unrelated to the PR topic

Comment (thadguidry) [reply .vscode/settings.json:2]: did not, let me fix.

**Review by wetneb — APPROVED** [review]
